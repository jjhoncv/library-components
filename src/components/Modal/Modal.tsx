import * as React from 'react';
import * as PropTypes from 'prop-types';
import { useState, useEffect, useRef } from 'react';

import {
  ModalOverlay,
  ModalStyled,
  ModalIcon
} from './styled';

import { Icon } from './../Icon';

interface Props {
  children?: React.ReactNode;
  openModal?: boolean;
  closeModal?: Function;
}

export const Modal: React.FC<Props> = ({
  openModal,
  children,
  closeModal
}) => {

  const [ open, setOpen ] = useState(openModal);
  const ref = useRef(null);

  useEffect(() => {
    setOpen(openModal);
  }, [openModal]);

  const handleClick = (e) => {
    if(e.currentTarget === e.target || ref.current.contains(e.target)) {
      setOpen(false);
      closeModal();
    }
  }

  return(
    <>
      {
        open && (
          <ModalOverlay onClick={(e) => handleClick(e)}>
            <ModalStyled>
              <ModalIcon
                ref={ref}
                onClick={(e) => handleClick(e)}>
                <Icon name="x"/>
              </ModalIcon>
              {children}
            </ModalStyled>
          </ModalOverlay>
        )
      }
    </>
  );
}

Modal.propTypes = {
  children: PropTypes.node,
  openModal: PropTypes.bool,
  closeModal: PropTypes.func
}
