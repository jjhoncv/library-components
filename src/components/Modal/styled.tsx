import styled from 'styled-components';

import { COLORS, FONTS } from './../../ui';

export const ModalOverlay = styled.div`
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  display: flex;
  opacity: 1;
  z-index: 100;
  padding: 15px;
  position: fixed;
  animation: show 0.5s ease;
  overflow-x: hidden;
  overflow-y: auto;
  align-items: center;
  justify-content: center;
  background-color: rgba(0,0,0,0.7);
`;
export const ModalStyled = styled.div`
  display: flex;
  flex-wrap: wrap;
  flex-direction: column;
  background: ${COLORS.WHITE};
  padding: 10px;
  color: black;
  position: relative;
  padding: 30px;
`;

export const ModalIcon = styled.div`
  position: absolute;
  top: 10px;
  right: 10px;
  display: inline-flex;
  align-self: flex-end;
  box-sizing: border-box;
  cursor: pointer;
  border-radius: 50%;
  transition: .3s linear;

  &:hover {
    background-color: ${COLORS.BORDER_GRAY};
  }

  svg {
    color: ${COLORS.BLACK};
  }
`;
