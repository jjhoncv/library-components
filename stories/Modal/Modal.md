# Modal

Importaci√≥n del componente Modal.

```js
import { Modal } from 'library/components/Modal';
import { Button } from 'library/components/Button';
```

<!-- STORY -->

#### Uso

```jsx
const Wrapper = () => {

  const [ open, setOpen ] = React.useState(false);

  return(
    <>
      <Button onClick={() => setOpen(true)} primary>Open Modal</Button>
      <Modal
        openModal={open}
        closeModal={() => setOpen(false)}
      >
        <div>
          <h1>Brief</h1>
          <h2>Nueva Plataforma Call Center</h2>
          <h4>library</h4>
        </div>
      </Modal>
    </>
  );
};

<Wrapper />
```
