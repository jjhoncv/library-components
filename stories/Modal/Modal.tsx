import * as React from 'react';
import { Modal } from '../../src/components/Modal';
import { Button } from '../../src/components/Button';
import { WithPropTypes } from '../../.storybook/Containers/WithPropTypes';

const Wrapper = () => {

  const [ open, setOpen ] = React.useState(false);

  return(
    <>
      <Button onClick={() => setOpen(true)} primary>Open Modal</Button>
      <Modal
        openModal={open}
        closeModal={() => setOpen(false)}
      >
        <div>
          <h1>Brief</h1>
          <h2>Nueva Plataforma Call Center</h2>
          <h4>library</h4>
        </div>
      </Modal>
    </>
  );
};

export default () => {
  return WithPropTypes({
    Component: Modal,
    RenderComponent: Wrapper,
    showProp: false
  });
}
