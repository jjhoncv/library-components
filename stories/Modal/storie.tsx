import { storiesOf } from '@storybook/react';
import Modal from './Modal';
import * as readme from './Modal.md';

const stories = storiesOf('2. Components/Modal', module);

stories.add('Default', Modal(),{
  readme: { content: readme }
});

