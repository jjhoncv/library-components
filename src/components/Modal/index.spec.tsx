import * as React from 'react';
import { shallow } from 'enzyme';
import { Modal } from './Modal';

describe('Render component', () => {
  describe('Modal => render snapshot', () => {
    it('Render correctly', () => {
      const wrapper = shallow(
        <Modal openModal={true}><div>test</div></Modal>
      );
      expect(wrapper.html()).toMatchSnapshot();
    })
  });

  describe('Modal Builder', () => {
    it('Modal => render children', () => {
      const wrapper = shallow(
        <Modal openModal={true}>
          <div>test</div>
        </Modal>);
      expect(wrapper.find('div').length).toBe(1);
    })
  })
})

