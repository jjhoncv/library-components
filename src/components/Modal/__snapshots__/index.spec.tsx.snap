// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render component Modal => render snapshot Render correctly 1`] = `"<div class=\\"sc-bdVaJa fPAUKf\\"><div class=\\"sc-bwzfXH hvcgaO\\"><div class=\\"sc-htpNat bnKPqT\\"><div class=\\"sc-bxivhb dGAJNz\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"><line x1=\\"18\\" y1=\\"6\\" x2=\\"6\\" y2=\\"18\\"></line><line x1=\\"6\\" y1=\\"6\\" x2=\\"18\\" y2=\\"18\\"></line></svg></div></div><div>test</div></div></div>"`;
